src/Network.cpp

```diff
@@ -860,9 +860,38 @@ Network::Netresult Network::get_scored_moves(
     if (ensemble == DIRECT) {
         assert(rotation >= 0 && rotation <= 7);
         result = get_scored_moves_internal(state, planes, rotation);
+    } else if (ensemble == MULTI_AVG && rotation > 1) {
+        // MULTI_AVG rotation is the # of random rotations to provide
+        assert(rotation >= 1 && rotation <= 8);
+        std::vector<int> r_list{0, 1, 2, 3, 4, 5, 6, 7};
+        // Need to account for extra 2 in m_squaresize.
+        std::vector<float> policy((BOARD_SIZE + 2) * (BOARD_SIZE + 1));
+
+        if (rotation < 8) {
+            std::shuffle(r_list.begin(), r_list.end(), Random::get_Rng());
+        }
+
+        result = get_scored_moves_internal(state, planes, r_list[0]);
+
+        for (int r = 1; r < rotation; ++r) {
+            Netresult tmpresult = get_scored_moves_internal(state, planes, r_list[r]);
+            result.second += tmpresult.second;
+
+            for (auto res : tmpresult.first) {
+                // +1 to account for PASS
+                policy[res.second + 1] += res.first;
+            }
+        }
+
+        for (size_t c = 0; c < result.first.size(); c++) {
+            result.first[c].first += policy[result.first[c].second + 1];
+            result.first[c].first /= rotation;
+        }
+
+        result.second /= rotation;
     } else {
-        assert(ensemble == RANDOM_ROTATION);
-        assert(rotation == -1);
+        assert((ensemble == RANDOM_ROTATION && rotation == -1) ||
+               (ensemble == MULTI_AVG && rotation == 1));
```

Comment:
This looks wrong to me: MULTI_AVG is handled in the other branch, so if ensemble is MULTI_AVG here something is seriously broken.

Reply:
It's true that MULTI_AVG with 1 rotation is effectively picking out a single rotation just like RANDOM_ROTATION. But what's the intent of allowing a subset of random rotation / `num_rotations` to be averaged? Why not always just average all 8? This would simplify the MULTI_AVG branch if `num_rotations` is assumed to be 8.

Reply:
I guess I just didn't want to tear the support for any # of rotations out since I bothered to write it originally lol. :)

Reply:
I updated this to use a num_rotations variable and so forth to try and make it more clear. I didn't seem to update this review automatically, but some changes have been made. Let me know if you need anything further?

Reply:
I'm with @Mardak on this one in that supporting multiple rotations averaged complicates the code for no gain, so I'd just get rid of that.